browser/themes/BUILD.gn

```diff
@@ -1,41 +1,32 @@
 source_set("themes") {
-  # Remove when https://github.com/brave/brave-browser/issues/10656 is resolved
-  check_includes = false
   sources = [
     "brave_dark_mode_utils.h",
     "brave_dark_mode_utils_android.cc",
   ]
 
   deps = [
     "//base",
+    "//brave/common:pref_names",
+    "//brave/common:switches",
   ]
 
   if (!is_android) {
     sources += [
-      "brave_dark_mode_utils.cc",
       "brave_dark_mode_utils_internal.cc",
```

Comment:
I think whole sources/deps can be moved to `sources.gni` instead of spliting. WDYT?

Reply:
yes, I tried to do it first. //brave/browser has a dependency //brave/browser/themes. So, it needs to be kept.

Reply:
I think we can deprecate `themes` target.

Reply:
Do we use this splitting pattern commonly? If so, I'm fine.

Reply:
Below //brave/browser files have been including //brave/browser/themes/foo files. So, I'm not fully sure whether it's fine to remove the target.

  - //brave/browser/brave_browser_process_impl.cc,
  - //brave/browser/brave_local_state_pref.cc, 
  - //brave/browser/ui/views/inforbars/brave_wayback_machine_inforbar_contents_view.cc
  ...

And also, some brave modules have been using both BUILD.gn and sources.gni and it will be used more to fix the dependency violation.

 - //brave/renderer/sources.gni, BUILD.gn
 - //brave/browser/brave_wallet/sources.gni, BUILD.gn
 - //brave/browser/sources.gni, BUILD.gn
 ... 

Even, Brian also approved //brave/browser/importer - https://github.com/brave/brave-core/pull/6664